clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/MoveExecutor.scala

```diff
@@ -195,25 +195,17 @@ class MoveExecutor[TI <: TransferIndex](moveCommand: MoveCommand[TI])
 
   private def copyGoogleObject(source: String,
                                destination: String,
-                               ioUtil: IoUtil): Future[Unit] = {
```

Comment:
This was the culprit (and the same bug was in `deleteGoogleObject` below. These methods were returning `Future`s, but the caller didn't know that and was wrapping the call in another `Future`. The outer `Future` always succeeded because all it was doing was starting the inner `Future`.

These methods are synchronous so I chose to have them just return `Unit` instead of removing the `Future`-wrapping at the call sites.